### PR TITLE
fix(integration): test_extra_replica_cleanup

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#8871

#### What this PR does / why we need it:

Adjust the test case following the changes in https://github.com/longhorn/longhorn-manager/pull/2918. We implemented a precheck before creating a new Replica CR, so no new Replica will be created is no suitable node and disk are available.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

**Comment by c3y1huang** [comment]
To PASS the test case is dependent on the fix in https://github.com/longhorn/longhorn-manager/pull/2939 for the volume condition. However, this test case is ready to review.

File: manager/integration/tests/test_ha.py [hunk 1]
```diff
@@ -1932,21 +1932,20 @@ def test_rebuild_after_replica_file_crash(client, volume_name): # NOQA
...
     3. Attach the volume to a node and write some data to it and
     save the checksum.
     4. Increase the volume replica number to 4.
-    5. Volume should show failed to schedule and an extra stop replica.
-    6. Decrease the volume replica number to 3.
-    7. Volume should show healthy and the extra failed to scheduled replica
-    should be removed.
-    8. Check the data in the volume and make sure it's same as the checksum.
+    5. No Replica should be created.
+    7. Volume should show failed to schedule.
```
Comment (roger-ryao) [inline manager/integration/tests/test_ha.py:1945]: LGTM, but there is a small error with the numbering of the cases.

**Comment by c3y1huang** [comment]
Test result: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7106/

**Review by roger-ryao — APPROVED** [review]
LGTM

**Pull request merged**